File: schemas/event-management-ticketing/mongodb/speakers.js
// speakers: Individuals who present at event sessions.
// See README.md for full design rationale.

const mongoose = require("mongoose");

const speakersSchema = new mongoose.Schema(
  {
    user_id: { type: mongoose.Schema.Types.ObjectId, ref: "User", default: null },
    name: { type: String, required: true },
    email: { type: String, default: null },
    bio: { type: String, default: null },
    title: { type: String, default: null },
    company: { type: String, default: null },
    avatar_url: { type: String, default: null },
    website_url: { type: String, default: null },
    twitter_handle: { type: String, default: null },
    linkedin_url: { type: String, default: null },
    is_active: { type: Boolean, required: true, default: true },
  },
  { timestamps: { createdAt: "created_at", updatedAt: "updated_at" } }
);

speakersSchema.index({ user_id: 1 });
speakersSchema.index({ is_active: 1 });

module.exports = mongoose.model("Speaker", speakersSchema);

File: schemas/event-management-ticketing/spacetimedb/speakers.rs
// speakers: Individuals who present or participate in event sessions.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = speakers, public)]
pub struct Speaker {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub user_id: Option<String>, // UUID — FK → users.id (set null)

    pub name: String,
    pub email: Option<String>,
    pub bio: Option<String>,
    pub title: Option<String>,
    pub company: Option<String>,
    pub avatar_url: Option<String>,
    pub website_url: Option<String>,
    pub twitter_handle: Option<String>,
    pub linkedin_url: Option<String>,
    pub is_active: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
    // Index: is_active
}
